src/server/scripts/EasternKingdoms/Stratholme/boss_baroness_anastari.cpp

```diff
@@ -88,37 +97,73 @@ class boss_baroness_anastari : public CreatureScript
             if (!UpdateVictim())
                 return;
 
-            //BansheeWail
-            if (BansheeWail_Timer <= diff)
-            {
-                if (rand32() % 100 < 95)
-                    DoCastVictim(SPELL_BANSHEEWAIL);
-                //4 seconds until we should cast this again
-                BansheeWail_Timer = 4000;
-            } else BansheeWail_Timer -= diff;
-
-            //BansheeCurse
-            if (BansheeCurse_Timer <= diff)
-            {
-                if (rand32() % 100 < 75)
-                    DoCastVictim(SPELL_BANSHEECURSE);
-                //18 seconds until we should cast this again
-                BansheeCurse_Timer = 18000;
-            } else BansheeCurse_Timer -= diff;
-
-            //Silence
-            if (Silence_Timer <= diff)
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
             {
-                if (rand32() % 100 < 80)
-                    DoCastVictim(SPELL_SILENCE);
-                //13 seconds until we should cast this again
-                Silence_Timer = 13000;
-            } else Silence_Timer -= diff;
+                if (invisible)
+                {
+                    if (!possessedTarget->HasAura(SPELL_POSSESSED))
+                        _events.ScheduleEvent(EVENT_INVISIBLE, Seconds(0));
+
+                    if (possessedTarget->GetHealthPct() <= 50)
```

Comment:
is this supposed to be a "else if" to avoid scheduling the same event twice ?

Reply:
You're right, I will test with OR to see if it also works.